stl/inc/xmemory

```diff
@@ -776,6 +776,30 @@ _NODISCARD constexpr allocation_result<typename allocator_traits<_Alloc>::pointe
 // The number of user bytes a single byte of ASAN shadow memory can track.
 _INLINE_VAR constexpr size_t _Asan_granularity = 8;
 
+_NODISCARD inline const void* _Get_asan_aligned_first(const void* const _First, const void* const _Last) noexcept {
+    const auto _First_address = reinterpret_cast<uintptr_t>(_First);
+    const auto _Last_address  = reinterpret_cast<uintptr_t>(_Last);
+
+    const auto _Aligned_address = (_First_address + _Asan_granularity - 1) & ~(_Asan_granularity - 1);
+
+    if (_Aligned_address > _Last_address) {
+        return nullptr;
+    } else {
+        return reinterpret_cast<const char*>(_Aligned_address);
+    }
+}
+
+// Works for standard containers with the `value_type` typedef.
+template <class _Container, class = void>
+_INLINE_VAR constexpr size_t _Container_allocation_minimum_alignment = alignof(typename _Container::value_type);
+
+template <class _Container>
+_INLINE_VAR constexpr size_t _Container_allocation_minimum_alignment<_Container,
+    void_t<decltype(_Container::allocator_type::_Minimum_allocation_alignment)>> =
+    alignof(typename _Container::value_type) > _Container::allocator_type::_Minimum_allocation_alignment
+        ? alignof(typename _Container::value_type)
+        : _Container::allocator_type::_Minimum_allocation_alignment;

```

Comment:
Could we use `_Max_value` instead of the conditional operator here?

Reply:
Yes, I'm the person that counts the number of template instantiations and alias expansions to evaluate the complexity of metaprogramming, so this suggestion may seem out of character. I think we all have at least a mild dislike of the conditional operator, especially when used to implement min / max which requires repeating everything twice, and the names we repeat here are not short. This reaches my threshold of "I'm willing to pay a few microseconds of compile time to not have to read this."